packages/@remirror/core/src/builtins/persistent-selection-extension.ts

```diff
@@ -0,0 +1,50 @@
+// XXX: @remirror/core exports a _type_ 'DecorationSet', but we need access to the static methods as well.
+// See https://github.com/iview/iview/issues/4465 for a similar issue
+import { ExtensionPriority } from '@remirror/core-constants';
+import type { Static } from '@remirror/core-types';
+import { Decoration, DecorationSet } from '@remirror/pm/view';
+
+import { extensionDecorator } from '../decorators';
+import { PlainExtension } from '../extension';
+import type { CreatePluginReturn } from '../types';
+
+export interface PersistentSelectionOptions {
+  persistentSelectionClass?: Static<string>;
+}
+
+@extensionDecorator<PersistentSelectionOptions>({
+  defaultOptions: {
+    persistentSelectionClass: undefined,
+  },
+  defaultPriority: ExtensionPriority.Medium,
+  staticKeys: ['persistentSelectionClass'],
+})
+export class PersistentSelectionExtension extends PlainExtension<PersistentSelectionOptions> {
+  get name() {
+    return 'persistentSelection' as const;
+  }
+
+  createPlugin(): CreatePluginReturn {
+    return {
+      props: {
+        decorations: ({ doc, selection }) => {
+          // Do not run the extension at all when there is no actual decoration
+          // to be done.
+          if (!this.options.persistentSelectionClass) {
+            return DecorationSet.empty;
+          }
+
+          if (selection.empty) {
```

Comment:
Now I'm not even sure how the previous test passes 😄 
It looks like you return early for empty selections.

Reply:
Oh I see, it passes because it's never run. The `setTimeout` means that the error will be thrown after the test.